CRM/Sqltasks/Action/APICall.php

```diff
@@ -202,6 +208,12 @@ public function execute() {
         }
       }
 
+      if ($store_api_results) {
+        $record_id = $query->id;
+        $result_json = json_encode($result);
+        CRM_Core_DAO::executeQuery("UPDATE `$data_table` SET `api_result` = '$result_json' WHERE `id` = $record_id");
```

Comment:
This will fail if the JSON payload contains an `'` symbol (and is technically an SQL injection); we should use placeholders and the second parameter of `executeQuery()` to provide the values.